nscplugin/src/main/scala/scala/scalanative/nscplugin/NirCodeGen.scala

```diff
@@ -76,11 +76,24 @@ abstract class NirCodeGen
   }
 
   class SaltyCodePhase(prev: Phase) extends StdPhase(prev) {
-    val curLocalInfo = new util.ScopedVar[CollectLocalInfo]
-    val curClassSym  = new util.ScopedVar[Symbol]
-    val curMethodSym = new util.ScopedVar[Symbol]
-    val curEnv       = new util.ScopedVar[Env]
-    val curThis      = new util.ScopedVar[Val]
+    private val curLocalInfo = new util.ScopedVar[CollectLocalInfo]
+    private val curClassSym  = new util.ScopedVar[Symbol]
+    private val curMethodSym = new util.ScopedVar[Symbol]
+    private val curEnv       = new util.ScopedVar[Env]
+    private val curThis      = new util.ScopedVar[Val]
+
+    private val lazyAnonDefs = mutable.Map.empty[Symbol, ClassDef]
```

Comment:
Reminds me of https://github.com/scala-js/scala-js/issues/2222 and its fix https://github.com/scala-js/scala-js/commit/533da726f8aad590af2692107e901c7f9bc234ea :p